frontend/app/services/SubscriptionService.scala

```diff
@@ -108,11 +108,15 @@ class SubscriptionService(val zuoraSoapClient: soap.ClientWithFeatureSupplier,
   implicit private val _bt = bt
 
   val membershipCatalog: FutureSupplier[MembershipCatalog] = new FutureSupplier[MembershipCatalog](
-    for {
-      zuoraCatalog <- zuoraRestClient.productCatalog
-    } yield MembershipCatalog.unsafeFromZuora(ratePlanIds)(zuoraCatalog.products.flatMap(_.productRatePlans))
+    productRatePlans.map(MembershipCatalog.unsafeFromZuora(ratePlanIds))
   )
 
+  def productRatePlans: Future[Seq[rest.ProductRatePlan]] =
+     zuoraRestClient.productCatalog.map(_.products.flatMap(_.productRatePlans))
+
+  def makeMembershipCatalog(): Future[MembershipCatalog.Val[MembershipCatalog]] =
```

Comment:
Not sure about this name. `makeMembershipCatalog` suggest some side effects, like persisting a record or similar...